Source/effects.cpp

```diff
@@ -1410,4 +1410,37 @@ void effects_play_sound(const char *snd_file)
 	}
 }
 
+typedef struct waveformat_tag {
+	WORD	wFormatTag;
+	WORD	nChannels;
+	DWORD	nSamplesPerSec;
+	DWORD	nAvgBytesPerSec;
+	WORD	nBlockAlign;
+} WAVEFORMAT;
+
+typedef struct pcmwaveformat_tag {
+	WAVEFORMAT	wf;
+	WORD		wBitsPerSample;
+} PCMWAVEFORMAT;

```

Comment:
This structure doesn't seem to exist under SDL? Makes no sense, how else are you supposed to read information about a WAV file?

Reply:
Audio is converted to output format on load, so you can check the output device for this info.

Reply:
I couldn't find anything on google. Are all the wave files initialized like they were in vanilla? We just need the seconds of the file (with at least 2 decimal points precision) or the header info to calculate it.

Reply:
From a game logic perspective yes, but wave.cpp has been replaced with SDL.

Would it be funny if Blizzard was dealing with this exact issue 14 years ago?
https://discourse.libsdl.org/t/time-length-of-sdl-mixer-chunks/12852/5

Reply:
:smile: